From 574de13faccbac52d4eda11d421ce39fcdb5a3d2 Mon Sep 17 00:00:00 2001
From: David Wu <david.wu@rock-chips.com>
Date: Thu, 25 Oct 2018 21:20:05 +0800
Subject: [PATCH] i2c: rk3x: Fix the smatch warning

This patch fixes the following warning:
drivers/i2c/busses/i2c-rk3x.c:1136 rk3x_i2c_xfer() error: uninitialized symbol 'timeout'.

Fixes: d5635ca05b11 ("i2c: rk3x: Disable irq after i2c transfer finished")
Change-Id: I7618ae660a62e8e3fc5b7b5d00cff1264bd18663
Signed-off-by: David Wu <david.wu@rock-chips.com>
---
 drivers/i2c/busses/i2c-rk3x.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/drivers/i2c/busses/i2c-rk3x.c b/drivers/i2c/busses/i2c-rk3x.c
index d0490b1b21ce..8a1adb5adf12 100644
--- a/drivers/i2c/busses/i2c-rk3x.c
+++ b/drivers/i2c/busses/i2c-rk3x.c
@@ -1132,10 +1132,7 @@ static int rk3x_i2c_xfer(struct i2c_adapter *adap,
 		}
 	}
 
-	/* Interrupt is disabled when i2c transfer is timeout */
-	if (timeout != 0)
-		rk3x_i2c_disable_irq(i2c);
-
+	rk3x_i2c_disable_irq(i2c);
 	rk3x_i2c_disable(i2c);
 
 	clk_disable(i2c->pclk);
-- 
2.35.3

